packages/plugin-lighthouse/src/lib/lighthouse-plugin.integration.test.ts

```diff
@@ -1,12 +1,48 @@
 import { expect } from 'vitest';
-import { pluginConfigSchema } from '@code-pushup/models';
-import { lighthousePlugin } from './lighthouse-plugin';
+import { AuditOutput, pluginConfigSchema } from '@code-pushup/models';
+import { getLogMessages } from '@code-pushup/test-utils';
+import { ui } from '@code-pushup/utils';
+import { getRunner, lighthousePlugin } from './lighthouse-plugin';
 
 describe('lighthousePlugin', () => {
   it('should create valid plugin config', () => {
-    const pluginConfig = lighthousePlugin('https://code-pushup-portal.com');
+    const pluginConfig = lighthousePlugin('https://www.google.com/');
     expect(() => pluginConfigSchema.parse(pluginConfig)).not.toThrow();
     expect(pluginConfig.audits.length).toBeGreaterThan(100);
     expect(pluginConfig.groups).toHaveLength(5);
   });
 });
+
+describe('getRunner', () => {
+  it('should create and execute runner correctly', async () => {
+    const runner = getRunner('https://www.google.com/', {
```

Comment:
I believe this is a bug we have in models. Will get back.

Reply:
The `getRunner` returns a `RunnerFunction` which is correct.

```ts
export function getRunner(
  urlUnderTest: string,
  flags: Flags = {},
): RunnerFunction { ... }
```

The problem highlighted in the IDE is related to the `zod` schema `runnerFunctionSchema`. To be more specific to the `onProgressSchema`. For some reason even if we defined the this argument as optional, the IDE highlights is. 

```ts

export const onProgressSchema = z
  .function()
  .args(z.unknown())
  .returns(z.void());
export type OnProgress = z.infer<typeof onProgressSchema>;

export const runnerFunctionSchema = z
  .function()
  .args(onProgressSchema.optional())
  .returns(z.union([auditOutputsSchema, z.promise(auditOutputsSchema)]));
```

However as the parsing of the `lighthousePlugin` `PluginConfig` does not throw the code should be correct.

Please let me know if you share that opinion and what we should do to go on.

Reply:
From what I found online, the reason seems to be that zod currently [does not support optional function arguments](https://github.com/colinhacks/zod/issues/2990).

How to proceed - not sure. The only way around it would currently be to switch to `RunnerConfig` which does not make sense in your use case. cc @matejchalk 

Side-note: Wondering if our progress strategy involving passing `onProgress` all the way is the best solution. This is an OOS discussion though.

Reply:
Given the lack of good Zod support, I guess the best we can do is pass in `undefined` (or `() => {}`) explicitly to avoid the type errors (`runner()` :arrow_right: `runner(undefined)`).

But I also don't actually understand what the `onProgress` parameter is for. Does it even need to be there?

Reply:
We could channel plugin specific progress logs. There are also tests for it to document it.

Reply:
added undefined